src/FffGcodeWriter.cpp

```diff
@@ -1588,6 +1526,81 @@ bool FffGcodeWriter::processSkinPart(const SliceDataStorage& storage, LayerPlan&
     return added_something;
 }
 
+void FffGcodeWriter::processSkinInsets(const SliceDataStorage& storage, LayerPlan& gcode_layer, const SliceMeshStorage& mesh, const int extruder_nr, const PathConfigStorage::MeshPathConfigs& mesh_config, const SkinPart& skin_part, unsigned int layer_nr, bool& added_something) const
+{
+    const int wall_0_extruder_nr = mesh.getSettingAsExtruderNr("wall_0_extruder_nr");
+    // add skin walls aka skin perimeters
+    if (extruder_nr == wall_0_extruder_nr)
+    {
+        for (const Polygons& skin_perimeter : skin_part.insets)
+        {
+            if (skin_perimeter.size() > 0)
+            {
+                added_something = true;
+                setExtruder_addPrime(storage, gcode_layer, layer_nr, extruder_nr);
+                gcode_layer.setIsInside(true); // going to print stuff inside print object
+                gcode_layer.addPolygonsByOptimizer(skin_perimeter, &mesh_config.insetX_config); // add polygons to gcode in inward order
+            }
+        }
+    }
+}
+
+void FffGcodeWriter::processSkinPartInfillGeneratePerimeterGaps(const SliceDataStorage& storage, LayerPlan& gcode_layer, const SliceMeshStorage& mesh, const int extruder_nr, const PathConfigStorage::MeshPathConfigs& mesh_config, const SkinPart& skin_part, unsigned int layer_nr, int skin_overlap, int skin_angle, EFillMethod pattern, const coord_t z, const bool generate_perimeter_gaps, Polygons& concentric_perimeter_gaps, bool& added_something) const
+{
+    const coord_t skin_line_width = mesh_config.skin_config.getLineWidth();
+    const int top_bottom_extruder_nr = mesh.getSettingAsExtruderNr("top_bottom_extruder_nr");
+
+    Polygons skin_polygons;
+    Polygons skin_lines;
+
+    const bool process_skin_infill = extruder_nr == top_bottom_extruder_nr;
```

Comment:
The same could be said for any if condition, but the prevailing consensus is:

    if (extruder_nr == top_bottom_extruder_nr)
    {
        processSkinInfill();
    }

rather than

    const bool process_skin_infill = extruder_nr == top_bottom_extruder_nr;
    if (process_skin_infill)
    {
        processSkinInfill();
    }

The first is much easier to read, right? It's also much more normal.